modules/tinymce/src/plugins/table/main/ts/actions/TableActions.ts

```diff
@@ -17,24 +17,29 @@ import { getCloneElements } from '../api/Settings';
 import { fireNewCell, fireNewRow } from '../api/Events';
 import Editor from 'tinymce/core/api/Editor';
 import { DomDescent } from '@ephox/phoenix';
+import { HTMLTableElement, Range } from '@ephox/dom-globals';
+
+export type BasicTableAction<T extends RunOperation.CombinedTargets = RunOperation.CombinedTargets> = (table: Element<HTMLTableElement>, target: T) => Option<Range>;
+export type PasteTableAction<T extends RunOperation.TargetPaste = RunOperation.TargetPaste> = (table: Element<HTMLTableElement>, target: T) => Option<Range>;
+export type AdvancedPasteTableAction<T extends RunOperation.TargetPasteRows = RunOperation.TargetPasteRows> = (table: Element<HTMLTableElement>, target: T) => Option<Range>;
```

Comment:
This can be simplified to just:
```
export type PasteTableAction = BasicTableAction<RunOperation.TargetPaste>;
export type AdvancedPasteTableAction = BasicTableAction<RunOperation.TargetPasteRows>;
```

Reply:
Hm. It complains about this. I assumed because TargetPaste doesn't extend CombinedTargets, and TargetPasteRows could but doesn't atm?

Reply:
Hmm... something is wrong with the types there then.

Reply:
At the very least we should at simplify this/remove duplication, so it's something like this:
```
type BasicTableAction<T> = (table: Element<HTMLTableElement>, target: T) => Option<Range>;
export TableAction = BasicTableAction<RunOperation.CombinedTargets>;
export type PasteTableAction = BasicTableAction<RunOperation.TargetPaste>;
export type AdvancedPasteTableAction = BasicTableAction<RunOperation.TargetPasteRows>;
```

Reply:
Discussed on slack/video call... my original suggestion was incorrect, so we should just go with the above simplification instead.